agent/php_user_instrument_llist.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if LOOKUP_METHOD == LOOKUP_USE_LINKED_LIST
+
+/*
+ * Purpose : Store zend_function's metadata (filename, lineno) in wraprec
+ *           for use in looking up the instrumentation.
+ * 
+ * Params  : 1. The wraprec to update with metadata
+ *           2. The zend function with meta data
+ *
+ * Returns : void
+ */
+static void wraprec_metadata_set(nruserfn_t* wraprec, zend_function* func) {
+    const char* filename = nr_php_function_filename(func);
+  /*
+   * Before setting a filename, ensure it is not NULL and it doesn't equal the
+   * "special" designation given to funcs without filenames. If the function is
+   * an evaluated expression or called directly from the CLI there is no
+   * filename, but the function says the filename is "-". Avoid setting in this
+   * case; otherwise, all the evaluated/cli calls would match.
+   */
+  if ((NULL == wraprec->filename) && (NULL != filename)
+      && (0 != nr_strcmp("-", filename))) {
+    wraprec->filename = nr_strdup(filename);
+  }
+
+  wraprec->lineno = nr_php_zend_function_lineno(func);
+
+  if (chk_reported_class(func, wraprec)) {
+    wraprec->reportedclass = nr_strdup(ZSTR_VAL(func->common.scope->name));
+  }
+}
+
+/*
+ * Purpose : Determine if a func matches a wraprec.
+ *
+ * Params  : 1. The wraprec to match to a zend function
+ *           2. The zend function to match to a wraprec
+ *
+ * Returns : True if the class/function of a wraprec match the class function
+ *           of a zend function.
+ */
+static inline bool nr_php_wraprec_matches(nruserfn_t* p, zend_function* func) {
```

Comment:
This code is a verbatim copy of https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_user_instrument.c#L322-L403. It's been factored out into lookup method implementation specific module for easier testing.